### ISSUE TITLE
Add a new project `turms-plugin-s3` to support storage service

### ISSUE THREAD

**Issue by JamesChenX** [turms-im/turms#1109]

**Comment by itmagus** [comment]
这是要新加一个插件吗？

**Comment by hongfanmeng** [comment]
Can I just use the `turms-plugin-minio` plugin and set the endpoint and access-key settings to the aws s3 settings?

**Comment by JamesChenX** [comment]
@hongfanmeng No, though Minio said that it is compatible with AWS S3, I have tried to use `io.minio.minio` to connect to the S3 service, but it has many problems and cannot work at all in many cases, so it's buggy and impractical to use `io.minio.minio` for the S3 service.

But instead of introducing a new project `turms-plugin-s3`, I think there is a better way to implement this feature: Rename the current project `turms-plugin-minio` to `turms-plugin-storage`, and provide different maven profiles in `turms-plugin-storage` for users to import different storage dependencies for Minio, S3, and etc, and implement different storage client based on the dependencies, so that we can share a lot of logic.
